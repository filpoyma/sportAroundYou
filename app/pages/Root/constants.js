export const ROUTE = {
  ATHLETE: 'athlete',
  TEST: {
    ACCELERATION_DIRECT_STANCE: 'acceleration_direct_stance',
    ACCELERATION_DIRECT_STANCE_INPUT_FORM: 'acceleration_direct_stance_input_form',
    ACCELERATION_SIDE_STANCE: 'acceleration_side_stance',
    ACCELERATION_SIDE_STANCE_INPUT_FORM: 'acceleration_side_stance_input_form',
    ANTHROPOMETRY_INPUT_FORM: 'anthropometry_input_form',
    ANTHROPOMETRY: 'anthropometry',
    ANXIETY: 'anxiety',
    ANXIETY_QUIZ: 'anxiety_quiz',
    BACK_FORWARD_RUN_INPUT_FORM: 'back_forward_run_input_form',
    BACK_FORWARD_RUN: 'back_forward_run',
    BEEP_INPUT_FORM: 'beep_input_form',
    BEEP: 'beep',
    CALIPEROMETRY_INPUT_FORM: 'caliperometry_input_form',
    CALIPEROMETRY: 'caliperometry',
    CALIPER_BODY_COMPOSITION: 'caliper_body_composition',
    CALIPER_BODY_COMPOSITION_INPUT_FORM: 'caliper_body_composition_input_form',
    COORDINATION_JUMPS_DIFF_VERTICAL: 'coordination_jumps_diff_vertical',
    COORDINATION_JUMPS_DIFF_HORIZONTAL: 'coordination_jumps_diff_horizontal',
    COORDINATION_JUMPS_DIFF_HORIZONTAL_INPUT_FORM: 'coordination_jumps_diff_horizontal_input_form',
    COORDINATION_TENNIS: 'coordination_tennis',
    COORDINATION_TENNIS_INPUT_FORM: 'coordination_tennis_input_form',
    CADENCE: 'cadence',
    CLAIMS: 'claims',
    CLAIMS_INPUT_FORM: 'claims_input_form',
    COPING_STRATEGIES_QUIZ: 'coping_strategies_quiz',
    COPING_STRATEGIES: 'coping_strategies',
    COMPETITIVE_ANXIETY: 'competitive_anxiety',
    COMPETITIVE_ANXIETY_QUIZ: 'competitive_anxiety_quiz',
    EXPLOSIVE_JUMP_INPUT_FORM: 'explosive_jump_input_form',
    EXPLOSIVE_JUMP: 'explosive_jump',
    EXPLOSIVE_THROW_INPUT_FORM: 'explosive_throw_input_form',
    EXPLOSIVE_THROW: 'explosive_throw',
    FLEXIBILITY_BACK_INPUT_FORM: 'flexibility_back_input_form',
    FLEXIBILITY_BACK: 'flexibility_back',
    FLEXIBILITY_SHOULDERS_INPUT_FORM: 'flexibility_shoulders_input_form',
    FLEXIBILITY_SHOULDERS: 'flexibility_shoulders',
    FMS: 'fms',
    FMS_INPUT_FORM: 'fms_input_form',
    HEMOGLOBIN_MASS: 'hemoglobin_mass',
    HEMOGLOBIN_MASS_INPUT_FORM: 'hemoglobin_mass_input_form',
    INBODY: 'inbody',
    ISOMED_ANKLE: 'isomed_ankle',
    ISOMED_ELBOW: 'isomed_elbow',
    ISOMED_HIP: 'isomed_hip',
    ISOMED_KNEE: 'isomed_knee',
    ISOMED_SHOULDER: 'isomed_shoulder',
    JUMPS_FATIGUE: 'jumps_fatigue',
    MAM: 'mam',
    MAM_RELATIVE: 'mam_relative',
    MAM_SKI_EGROMETER: 'mam_ski_egrometer',
    MAM_SKI_EGROMETER_INPUT_FORM: 'mam_ski_egrometer_input_form',
    MOTILITY_JUMPS_DIFF_HORIZONTAL_INPUT_FORM: 'motility_jumps_diff_horizontal_input_form',
    MOTILITY_JUMPS_DIFF_HORIZONTAL: 'motility_jumps_diff_horizontal',
    MOTILITY_JUMPS_DIFF_VERTICAL: 'motility_jumps_diff_vertical',
    SPATIAL_INTELLECTION: 'spatial_intellection',
    SPATIAL_INTELLECTION_INPUT_FROM: 'spatial_intellection_input_from',
    RPE: 'rpe',
    RPE_QUIZ: 'rpe_quiz',
    RAM: 'ram',
    SPEED_AND_POWER_THREE_JUMPS: 'speed_and_power_three_jumps',
    SPIROERGOMETRY: 'spiroergometry',
    SPIROERGOMETRY_01051: 'spiroergometry 01.05.1',
    SPIROERGOMETRY_01052: 'spiroergometry 01.05.2',
    SPIROERGOMETRY_01061: 'spiroergometry 01.06.1',
    SPIROERGOMETRY_01062: 'spiroergometry 01.06.2',
    SPIROERGOMETRY_01071: 'spiroergometry 01.07.1',
    SPIROERGOMETRY_01072: 'spiroergometry 01.07.2',
    SPIROERGOMETRY_01081: 'spiroergometry 01.08.1',
    SPIROERGOMETRY_01082: 'spiroergometry 01.08.2',
    SPORT_MOTIVATION_QUIZ: 'sport_motivation_quiz',
    SPORT_MOTIVATION_SCALE_QUIZ: 'sport_motivation_scale_quiz',
    SPORT_MOTIVATION_SCALE: 'sport_motivation_scale',
    SPORT_MOTIVATION: 'sport_motivation',
    STRESS_AND_RECOVERY_QUIZ: 'stress_and_recovery_quiz',
    STRESS_AND_RECOVERY: 'stress_and_recovery',
    STRESS_FACTORS_QUIZ: 'stress_factors_quiz',
    STRESS_FACTORS: 'stress_factors',
    VISUAL_ATTENTION: 'visual_attention',
    VISUAL_ATTENTION_INPUT_FORM: 'visual_attention_input_form',
    TREMOR: 'tremor',
    TREMOR_INPUT_FORM: 'tremor_input_form'
  }
};
