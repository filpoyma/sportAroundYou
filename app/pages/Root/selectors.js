import { createSelector } from 'reselect';

const selectLocation = state => state.router?.location;

export const makeSelectLocation = () =>
  createSelector(
    selectLocation,
    location => location
  );
