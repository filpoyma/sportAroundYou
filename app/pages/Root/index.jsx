import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';

import Footer from '@/containers/Footer';
import Header from '@/containers/Header';
import GlobalStyles from '@/GlobalStyles';
import NotFoundPage from '@/pages/NotFoundPage/Loadable';
import TestPage from '@/pages/TestPage/Loadable';

import { ROUTE } from './constants';

const Root = () => (
  <>
    <Header />
    <main>
      <Switch>
        <Redirect exact from="/" to={`/${ROUTE.ATHLETE}/45`} />
        <Route path={`/${ROUTE.ATHLETE}/:athleteId`} component={TestPage} />
        <Route component={NotFoundPage} />
      </Switch>
    </main>
    <Footer />
    <GlobalStyles />
  </>
);

export default Root;
