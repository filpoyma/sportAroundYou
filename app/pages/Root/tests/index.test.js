import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Root from '../index';

const renderer = new ShallowRenderer();

describe('<Root />', () => {
  it('should shallow render and match the snapshot', () => {
    renderer.render(<Root />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
