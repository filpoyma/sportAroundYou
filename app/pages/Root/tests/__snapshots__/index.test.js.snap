// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> should shallow render and match the snapshot 1`] = `
<React.Fragment>
  <ForwardRef(HeaderContainer) />
  <main>
    <Switch>
      <Redirect
        exact={true}
        from="/"
        to="/77"
      />
      <Route
        component={[Function]}
        path="/:athleteId"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </main>
  <ForwardRef(FooterContainer) />
  <GlobalStyleComponent />
</React.Fragment>
`;
